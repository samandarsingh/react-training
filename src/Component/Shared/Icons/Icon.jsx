import React from "react";
import PropTypes from "prop-types";
import styles from "./Icon.module.scss";
/**
 * 
 * @param {string} fileName 
 * @param {string} altTitle
 * @return Element
 */
const Icon = props => {
    const {fileName,altTitle} = props;
    return (        
        <img className={styles.iconImage} 
            src={`assets/Icons/${fileName}`}
            alt={altTitle}>
        </img>       
    );
}
Icon.propTypes = {
    fileName: PropTypes.string.isRequired,
    altTitle: PropTypes.string.isRequired
}
export default Icon;
