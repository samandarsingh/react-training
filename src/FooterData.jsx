
const FooterData = 
    [
        {
            IsNewColumn: true,
            heading: "Customer Service",
            Items: [
            { title: "Help Center" },
            { title: "Contact Us" },
            { title: "Return an Item" },
            { title: "Recall Information" },
            { title: "Safety Data Sheets" }
            ]
        },
        {
            IsNewColumn: true,
            heading: "Corporate Info",
            Items: [
            { title: "Stapies Business Advantage Overview" },
            { title: "Industries Served" },
            { title: "Blog" },
            { title: "Our Brand" }
            ]
        },
        {
            IsNewColumn: true,
            heading: "Staples Corporate Solution",
            Items: [
            { title: "Office Products" },
            { title: "Technology Products" },
            { title: "Furniture" },
            { title: "Facilites" },
            { title: "Promotional Products" }
            ]
        },
        {
            IsNewColumn: true,
            heading: "New Customer",
            Items: [
            { title: "Become a Customer" },
            { title: "Govt. Customers" },
            { title: "Memberships" }
            ]
            },
        {
            IsNewColumn: false,
            heading: "Join the Staples Team",
            Items: [{ title: "Careers" }]
        }
    ];
    const footerSocialIcons = [
        { title: "Linkdin", file: "linkedin.png" },
        { title: "Twitter", file: "twitter.png" },
        { title: "Facebook", file: "facebook.png" },
        { title: "Youtube", file: "youtube.png" },
        { title: "Instagram", file: "insta.png" }
      ];     
    export {FooterData,footerSocialIcons};