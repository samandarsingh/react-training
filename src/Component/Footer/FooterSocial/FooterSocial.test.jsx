import React from "react";
import { mount } from "enzyme";
import Social from "./FooterSocial";
import Icon from "../../Shared/Icons";
import {footerSocialIcons} from "../../../FooterData"
describe(`my component testing`,()=> {     
    const wrapper = mount(<Social socialData={footerSocialIcons}/>);
    it(`my component should be run`,()=>{         
        expect(wrapper).toBeTruthy();
    });
    it(`passes child component lenght should be equal to 5`,()=> { 
        expect(wrapper.find(Icon)).toHaveLength(5); 
    }); 
    it(`getting file name should be equal`, () => {   
            
        expect(wrapper.find(".footerSocialIcon").childAt(0).find('img').prop('alt')).toEqual('Linkdin');
        expect(wrapper.find(".footerSocialIcon").childAt(1).find('img').prop('alt')).toEqual('Twitter');
        expect(wrapper.find(".footerSocialIcon").childAt(2).find('img').prop('alt')).toEqual('Facebook');
        expect(wrapper.find(".footerSocialIcon").childAt(3).find('img').prop('alt')).toEqual('Youtube');
        expect(wrapper.find(".footerSocialIcon").childAt(4).find('img').prop('alt')).toEqual('Instagram');
    }); 
    it(`renders correctly`, () => {   
        expect(wrapper).toMatchSnapshot();
    });  
});