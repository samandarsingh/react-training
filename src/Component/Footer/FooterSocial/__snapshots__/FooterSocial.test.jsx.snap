// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my component testing renders correctly 1`] = `ReactWrapper {}`;
